crates/router/src/routes/payments.rs

```diff
@@ -321,6 +322,14 @@ pub async fn payments_confirm(
         return http_not_implemented();
     };
 
+    if payload.customer_ip.is_none() {
+        payload.customer_ip = req
+            .headers()
+            .get("x-forwarded-for")
+            .and_then(|val| val.to_str().ok().and_then(|ip| ip.split(',').next()))
```

Comment:
Add the comments as to what this transformation does. Provide the original ip address format and the transformed one after running this logic